common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/validation/ValidRollingStartNumberValidator.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Corona-Warn-App
+ * SAP SE and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package app.coronawarn.server.common.persistence.domain.validation;
+
+import static java.time.ZoneOffset.UTC;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class ValidRollingStartNumberValidator implements ConstraintValidator<ValidRollingStartNumber, Long> {
+
+  @Override
+  public boolean isValid(Long rollingStartNumber, ConstraintValidatorContext constraintValidatorContext) {
+    long currentInstant = LocalDateTime.ofInstant(Instant.now(), UTC).toEpochSecond(UTC) / 600L;
+    return rollingStartNumber > 0L && rollingStartNumber < currentInstant;
```

Comment:
This still seems too broad to me. I have issues accepting keys from the 80s should be valid.

Reply:
[We don't](https://github.com/corona-warn-app/cwa-server/blob/feature/key-validation-distribution/services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java#L160): A key from the 80s can still be semantically valid - it will just not arrive in our database.

Reply:
I missed this. Sorry